adaptive_components/lib/src/adaptive_column.dart

```diff
@@ -0,0 +1,131 @@
+  /// top to bottom.
+  ///
+  /// To see an example visit:
+  /// https://adaptive-components.web.app/#/
+  const AdaptiveColumn({
+    this.gutter,
+    this.margin,
+    @required this.children,
+  }) : assert(children != null);
+
+  /// Empty space at the left and right of this widget.
+  final double margin;
+
+  /// Represents the space between children.
+  final double gutter;
+  final List<AdaptiveContainer> children;
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints constraints) {
+        BreakpointSystemEntry _entry = getBreakpointEntry(context);
+        final double _margin = margin ?? _entry.margin;
+        final double _gutter = gutter ?? _entry.gutter;
+        final int _totalColumns = _entry.columns;
+
+        return Container(
+          width: MediaQuery.of(context).size.width - (_margin * 2),
+          margin: EdgeInsets.symmetric(horizontal: _margin),
+          constraints: BoxConstraints(
+            minWidth: _entry.adaptiveWindowType.widthRangeValues.start,
+            maxWidth: _entry.adaptiveWindowType.widthRangeValues.end,
+          ),
+          child: Wrap(
+            runSpacing: 8.0,
+            children: () {
+              int currentColumns = 0;
+              int totalGutters = 0;
+              List<Widget> children = [];
+              List<AdaptiveContainer> row = [];
+
+              for (AdaptiveContainer child in this.children) {
+                // The if statement checks if the adaptiveContainer child fits
+                // within the adaptive constraints.
+                if (child.adaptiveConstraints
+                    .withinAdaptiveConstraint(context)) {
+                  row.add(child);
+                  currentColumns += child.adaptiveColumn;
+
+                  if (currentColumns < _totalColumns) {
+                    totalGutters++;
+                  } else {
+                    int gutters = 0;
+                    if (currentColumns > _totalColumns) {
+                      totalGutters--;
+                    }
+
+                    for (AdaptiveContainer rowItem in row) {
+                      // maxWidth equals column width without margin and gutters
+                      // divided by the total number of adaptive containers.
+                      double maxWidth = (MediaQuery.of(context).size.width -
+                              _margin * 2 -
+                              _gutter *
+                                  (gutters == totalGutters &&
+                                          currentColumns > _totalColumns
+                                      ? 0
+                                      : totalGutters)) /
+                          _totalColumns *
+                          rowItem.adaptiveColumn;
+                      children.add(
+                        ConstrainedBox(
+                          constraints: BoxConstraints(
```

Comment:
`BoxConstraints.tightFor` may be used here

Reply:
tightfor doesn't allow for min and max width. The min and max width comes from the breakpoints package. It is the range value.

Reply:
tightFor can take in a width, and then sets the minWidth and maxWidth to that value